wagtail/snippets/views/snippets.py

```diff
@@ -1120,8 +1163,21 @@ def snippets_model_check(app_configs, **kwargs):
 
         checks.register(snippets_model_check, "panels")
 
+    def register_model_methods(self):
+        @classmethod
+        def get_admin_url_namespace(cls):
+            return self.get_admin_url_namespace()
+
+        @classmethod
+        def get_admin_base_path(cls):
+            return self.get_admin_base_path()
+
+        self.model.get_admin_url_namespace = get_admin_url_namespace
+        self.model.get_admin_base_path = get_admin_base_path
```

Comment:
Do we want to keep doing this? Or should we advocate for `model.snippet_viewset.get_admin_url_namespace()` etc.?

It is quite handy, but I'm not really a fan of doing too much patching on the model, especially now that we have a reference to the viewset through `model.snippet_viewset`.

We don't have to remove this outright (in case people depend on it, even though these aren't documented), but I'd like us to have a preferred way to get these values.

Reply:
I'm very much in favour of keeping as much of this as possible contained within the viewset class - patching the model with these methods seems like it was a necessary evil before we had an obvious place other than the model to hang this common functionality onto, but the viewset is the natural place for it now. I think it's reasonable to remove it outright - I can't imagine there being any expectation for these to be anything more than Wagtail-internal glue.

In an ideal world I'd like to see the `snippet_viewset` attribute dropped too, so that the view-level and model-level considerations are fully separate - but it sounds like it'd need some heavy refactoring of the bulk action views to make that happen, and patching a single `snippet_viewset` attribute is definitely a lot cleaner than patching multiple methods into the model.

Reply:
I think I'll do this in a separate PR, as the `get_admin_url_namespace` is still used in a few places in Wagtail.

> In an ideal world I'd like to see the `snippet_viewset` attribute dropped too, so that the view-level and model-level considerations are fully separate - but it sounds like it'd need some heavy refactoring of the bulk action views to make that happen, and patching a single `snippet_viewset` attribute is definitely a lot cleaner than patching multiple methods into the model.

We might be able to get away with something simpler to achieve this, e.g. the idea in https://github.com/wagtail/wagtail/pull/10178#discussion_r1135401704:

> Or something simpler, we could change the `SNIPPET_MODELS` global list into a dictionary that maps each model to the viewset instance? People who rely on `get_snippet_models()` will still be able to iterate through it like a list of models, but we will be able to do something like `get_snippet_models()[MyModel]` and get the viewset. Still, it's not as handy compared to `MyModel.snippet_viewset`, though.

Or introduce another method, e.g.

```py
def get_snippet_viewset(model):
    return get_snippet_models()[model]
```